src/utils.ts

```diff
@@ -1,9 +1,20 @@
 import { keccak256 } from "@ethersproject/solidity"
 import { IncrementalMerkleTree, MerkleProof } from "@zk-kit/incremental-merkle-tree"
-import { poseidon } from "circomlibjs"
+import { buildPoseidonOpt } from "circomlibjs"
 import { ZqField } from "ffjavascript"
 import { StrBigInt } from "./types"
 
+/**
+ *  Wrapper for Poseidon hash function to parse return value format from Uint8Array to BigInt
+ */
+export async function buildPoseidon() {
+  const poseidon = await buildPoseidonOpt()
+
+  return (input: any): bigint => {
+    const result = poseidon(input)
+    return BigInt(poseidon.F.toString(result))
```

Comment:
To fix encoding issue with latest version of circomlibjs, I made wrapper function for buildPoseidonOpt that returns a parsed BigInt value.
With this way, all the test cases run successfully, but I'm not sure if this is a best practice. 🤔